Jenkinsfile

```diff
@@ -393,49 +435,385 @@ def deploy(folderName, projName, openShiftHost, openShiftToken, mysqlUser, mysql
     """
 }
 
+def makePostRequest(url, body, method) {
 
+	println('service url...'+url);
+	println('post body...'+body)
 
+	def post = new URL(url).openConnection();
+	post.setRequestMethod(method)
+	post.setDoOutput(true)
+	post.setRequestProperty('Content-Type', 'application/json')
+	post.setRequestProperty('Accept', 'application/json')
+	post.getOutputStream().write(body.getBytes('UTF-8'))
+	def responseCode = post.getResponseCode();
+	if (responseCode != 200 && responseCode != 201) {
+		println('Failed. HTTP response: ' + responseCode)
+		println(post.getInputStream().getText());
+		assert false
+	} else {
+		println('Tested successfully!')
+		println(post.getInputStream().getText());
 
-def makeGetRequest(url) {
+	}
+}
 
-	println('service...'+url);
+//////// 3scale API Publishing ////////////////////////////
+import groovy.json.JsonOutput
+import groovy.json.JsonSlurperClassic 
+import javax.net.ssl.HostnameVerifier
+import javax.net.ssl.HttpsURLConnection
+import javax.net.ssl.SSLContext
+import javax.net.ssl.TrustManager
+import javax.net.ssl.X509TrustManager
+import java.security.SecureRandom
+import java.net.URLEncoder
+
```

Comment:
If possible, Groovy is better put in a separated file, and the pipeline uses the functions in this file.

Reply:
other supported way is to load/create a shared groovy library which contain all 3scale methods, but again it will be a maintenance issue and need an extra configuration steps in Jenkins. However will do R&D on that point to figure out how Jenkins load/import separate scripts file from same directory in a pipeline.